giotto/homology/preprocessing.py

```diff
@@ -0,0 +1,160 @@
+"""Local point cloud extractor"""
+# License: Apache 2.0
+
+
+import numpy as np
+from joblib import Parallel, delayed
+from scipy.spatial.distance import squareform, pdist
+from sklearn.base import BaseEstimator
+from sklearn.base import TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+from ..utils import validate_params
+
+
+def _check_k_min_max(k_min, k_max):
+    if k_min > k_max:
+        raise ValueError("'k_min' should be less than 'k_max', but got {} and "
+                         "{}".format(k_min, k_max))
+
+
+class LocalNeighborhood(BaseEstimator, TransformerMixin):
+    """For each entry of ``X``, compute the (local) point cloud associated to
+    it by considering the neighborhood points according to the chosen
+    ``metric``.
+    The number of points per (local) point cloud is affected by three
```

Comment:
The interplay netween k_min, k_max and dist_percentage should be explained more clearly.